File: Lab2/Task1/LayoutExploration/app/src/main/java/com/example/layoutexploration/MainActivity.java
package com.example.layoutexploration;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button linearLayoutButton =
                findViewById(R.id.linearLayoutButton);
        linearLayoutButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this,
                    LinearLayoutActivity.class);
            startActivity(intent);
        });
        Button relativeLayoutButton =
                findViewById(R.id.relativeLayoutButton);
        relativeLayoutButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this,
                    RelativeLayoutActivity.class);
            startActivity(intent);
        });
        Button constraintLayoutButton =
                findViewById(R.id.constraintLayoutButton);
        constraintLayoutButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this,
                    ConstraintLayoutActivity.class);
            startActivity(intent);
        });
        Button gridLayoutButton =
                findViewById(R.id.gridLayoutButton);
        gridLayoutButton.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this,
                    GridLayoutActivity.class);
            startActivity(intent);
        });
    }}

File: Lab1/Task1/app/src/main/java/com/example/myfirstapp/MainActivity.kt
package com.example.myfirstapp

//import androidx.activity.enableEdgeToEdge

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        var button: Button
        var textView: TextView
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        textView = findViewById(R.id.textView)
        button = findViewById(R.id.button)
        button.setOnClickListener{
            textView.text="Button Clicked!"
            textView.setTextColor(Color.RED)
        }

        // Get reference to the "Go to Second Activity" button
        val buttonNavigate: Button = findViewById(R.id.buttonNavigate)
        val intent = Intent(this, SecondActivity::class.java)

        // Set a click listener for the button to navigate to SecondActivity
        buttonNavigate.setOnClickListener {
            startActivity(intent)
        }

        intent.putExtra("EXTRA_MESSAGE","Hello from MainActivity!");

        button.setOnClickListener {
            textView.text = "Button Clicked!"
            textView.setTextColor(Color.RED)

            // Show a Toast when the button is clicked
            Toast.makeText(this, "Button was clicked!", Toast.LENGTH_SHORT).show()
        }
    }

}